Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.h

```diff
@@ -255,28 +391,339 @@ class Wiimote : public ControllerEmu::EmulatedController
+    I2CBus i2c_bus;
+
+    // The port on the end of the motion plus:
+    ExtensionPort extension_port{i2c_bus};
+
+    void Update();
+
+#pragma pack(push, 1)
+    struct MotionPlusRegister
+    {
+      u8 controller_data[21];
+      u8 unknown[11];
+
+      // address 0x20
+      u8 calibration_data[0x20];
+      u8 unknown2[0xb0];
+
+      // address 0xF0
+      u8 initialized;
+
+      u8 unknown3[6];
+
+      // address 0xf7
+      // Wii Sports Resort reads regularly and claims mplus is disconnected if not to its liking
+      // Value starts at 0x00 and goes up after activation (not initialization)
+      // Immediately returns 0x02, even still after 15 and 30 seconds
+      u8 initialization_status;
+
+      u8 unknown4[2];
+
+      // address 0xFA
+      u8 ext_identifier[6];
+    } reg_data;
+#pragma pack(pop)
+
+    static_assert(0x100 == sizeof(reg_data));
+
+  private:
+    static const u8 INACTIVE_DEVICE_ADDR = 0x53;
+    static const u8 ACTIVE_DEVICE_ADDR = 0x52;
+
+    enum class PassthroughMode : u8
+    {
+      PASSTHROUGH_DISABLED = 0x04,
+      PASSTHROUGH_NUNCHUK = 0x05,
+      PASSTHROUGH_CLASSIC = 0x07,
+    };
+
+    // TODO: savestate
+    u8 times_updated_since_activation = 0;
+
+    bool IsActive() const { return ACTIVE_DEVICE_ADDR << 1 == reg_data.ext_identifier[2]; }
+
+    PassthroughMode GetPassthroughMode() const
+    {
+      return static_cast<PassthroughMode>(reg_data.ext_identifier[4]);
+    }
+
+    // TODO: when activated it seems the motion plus reactivates the extension
+    // It sends 0x55 to 0xf0
+    // It also writes 0x00 to slave:0x52 addr:0xfa for some reason
+    // And starts a write to 0xfa but never writes bytes..
+    // It tries to read data at 0x00 for 3 times (failing)
+    // then it reads the 16 bytes of calibration at 0x20 and stops
+
+    // TODO: if an extension is attached after activation, it also does this.
+
+    int BusRead(u8 slave_addr, u8 addr, int count, u8* data_out) override
+    {
+      if (IsActive())
+      {
+        // Motion plus does not respond to 0x53 when activated
+        if (ACTIVE_DEVICE_ADDR == slave_addr)
+          return RawRead(&reg_data, addr, count, data_out);
+        else
+          return 0;
+      }
+      else
+      {
+        if (INACTIVE_DEVICE_ADDR == slave_addr)
+          return RawRead(&reg_data, addr, count, data_out);
+        else
+        {
+          // Passthrough to the connected extension (if any)
+          return i2c_bus.BusRead(slave_addr, addr, count, data_out);
+        }
+      }
+    }
+
+    int BusWrite(u8 slave_addr, u8 addr, int count, const u8* data_in) override
+    {
+      if (IsActive())
+      {
+        // Motion plus does not respond to 0x53 when activated
+        if (ACTIVE_DEVICE_ADDR == slave_addr)
+        {
+          auto const result = RawWrite(&reg_data, addr, count, data_in);
+          return result;
+
+          // It seems a write of any value triggers deactivation.
+          if (0xf0 == addr)
```

Comment:
This if statement is after a return.

Reply:
Sorry.. leftovers from debugging. I've already removed it locally.